modules/data-streams/src/yamlRestTest/resources/rest-api-spec/test/data_stream/220_failure_store_cluster_setting.yml

```diff
@@ -0,0 +1,243 @@
+setup:
+  - requires:
+      reason: "Data stream options was added in 8.18+"
+      test_runner_features: [ capabilities, allowed_warnings, contains ]
+      capabilities:
+        - method: POST
+          path: /{index}/_doc
+          capabilities: [ 'failure_store_status' ]
+        - method: POST
+          path: /_index_template/{template}
+          capabilities: [ 'failure_store_in_template' ]
+
+  - do:
+      cluster.put_settings:
+        body:
+          persistent:
+            data_streams.failure_store.enabled: '*-matching'
+
+  - do:
+      ingest.put_pipeline:
+        id: "failing_pipeline"
+        body:  >
+          {
+            "processors": [
+              {
+                "fail": {
+                  "message" : "pipeline go boom"
+                }
+              }
+            ]
+          }
+
+  - do:
+      indices.put_index_template:
+        name: index_template_default_fs
+        body:
+          index_patterns: default-fs-*
+          data_stream: {}
+          template:
+            settings:
+              number_of_shards:   1
+              number_of_replicas: 1
+
+  - do:
+      cluster.put_component_template:
+        name: component_template_disabled_fs
+        body:
+          template:
+            data_stream_options:
+              failure_store:
+                enabled: false
+
+  - do:
+      indices.put_index_template:
+        name: index_template_disabled_fs
+        body:
+          index_patterns: disabled-fs-*
+          data_stream: {}
+          composed_of:
+            - component_template_disabled_fs
+          template:
+            settings:
+              number_of_shards:   1
+              number_of_replicas: 1
+
+---
+teardown:
+  - do:
+      indices.delete_data_stream:
+        name: default-fs-matching
+        ignore: 404
+
+  - do:
+      indices.delete_data_stream:
+        name: disabled-fs-matching
+        ignore: 404
+
+  - do:
+      indices.delete_index_template:
+        name: index_template_disabled_fs
+        ignore: 404
+
+  - do:
+      cluster.delete_component_template:
+        name: component_template_disabled_fs
+        ignore: 404
+
+  - do:
+      indices.delete_index_template:
+        name: index_template_default_fs
+        ignore: 404
+
+  - do:
+      ingest.delete_pipeline:
+        id: "failing_pipeline"
+        ignore: 404
+
+  - do:
+      cluster.put_settings:
+        body:
+          persistent:
+            data_streams.failure_store.enabled: ''
```

Comment:
Does it not accept `null` here? It might be better to completely reset it.

Reply:
Oh, yeah, that is neater — thanks.